src/component-library/CTA/BaseCTA.tsx

```diff
@@ -0,0 +1,40 @@
+import { forwardRef, HTMLAttributes } from 'react';
+import { StyledComponent } from 'styled-components';
+
+import { Sizes, Variants } from '../utils/prop-types';
+import { OutlinedCTA, PrimaryCTA, SecondaryCTA, StyledCTAProps } from './CTA.style';
+
+const ctaElements: Record<Variants, StyledComponent<'button', any, StyledCTAProps, never>> = {
+  primary: PrimaryCTA,
+  secondary: SecondaryCTA,
+  outlined: OutlinedCTA
+};
+
+type Props = {
+  variant?: Variants;
+  fullWidth?: boolean;
+  size?: Sizes;
+  disabled?: boolean;
+  as?: any;
```

Comment:
Why are we passing in the `as` prop?

Reply:
Its for the CTALink. We use it to pass `Link` from react-router.

Also, this BaseCTA wont be exported by the lib.